src/web/docusaurus/docs/tools-and-technologies/prisma.md

```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 11
+---
+
+# Prisma
+
+## Our Database
+
+[Supabase](./supabase.md) is one of the technologies that Telescope uses. Because it is built on top of Postgres, we have full access to a Postgres DB that can be used to store all sorts of data used by both frontend and backend.
+
+> PostgreSQL is a highly stable database backed by more than 20 years of development by the open-source community.
+
+## Database Schema
+
+Our database schema is defined in [schema.prisma](https://github.com/Seneca-CDOT/telescope/blob/master/src/db/prisma/schema.prisma). This is where we defined models and their relations using Prisma schema syntax.
+
+Managing the database schema using Prisma, we can track granular changes to the database schema. These granular changes are reflected as separate SQL script files. That way, our granular schema changes are reflected as code that can be captured with any version control software. This makes sure that the local development database or the production database is always up to date with the latest Schema.
+
+## Database migration
+
+Whenever we start up the DB container or after updating our schema, we should run a migration to apply new changes to the database if any.
+
+## Running migration
+
+```bash
+  cd src/db
+  cp env.example .env
+  pnpm migrate
+```
+
+See also [DB maintenance on staging and production](../contributing/database-maintenance.md).
+
+## Creating migration
+
+For a migration that has "narrowing" nature such as dropping a table, dropping a column or converting data types or similar changes, running a migration requires attention to make sure the data loss is under control. Prisma is smart enough to warn us about any potential data loss.
+
+### Schema change
+
+Simply edit the schema in [schema.prisma](https://github.com/Seneca-CDOT/telescope/blob/master/src/db/prisma/schema.prisma) and apply the migration.
+
+```bash
+  cd src/db
+  cp env.example .env
+  pnpm migrate
+```
+
+### Custom migration
+
+Sometimes, we want to make changes other than the modifying the schema, for example, adding new RLS to a table. We can do so by creating an empty migration file, which is an empty SQL script where we write our own SQL queries to be apply to the database.
```

Comment:
Can we use this to somehow include a migration for setting up the local db, but not run it by default in any environment other than dev?

I want to insert a user + feed based on #3501

Reply:
Yeah, I think it should be possible to have `INSERT` statements in the migration.

Reply:
I think all migrations will be included. To seed the DB, people often run a script.

Reply:
OK, so we need to do this outside of migrations.  We'll have to think about how to do it best, so we don't do it in tests, and we don't rely on users to do it locally (new users won't, it will be annoying to do it every time for regular devs).

Reply:
I'm thinking of putting the seed script can be `src/db`, making it read from a JSON file containing the example data. We call the script in `pnpm services:start`.